oci/layout/oci_dest.go

```diff
@@ -302,6 +305,33 @@ func (d *ociImageDestination) Commit(context.Context, types.UnparsedImage) error
 	return os.WriteFile(d.ref.indexPath(), indexJSON, 0644)
 }
 
+// PutBlobFromLocalFile arranges the data from path to be used as blob with digest.
+// It computes, and returns, the digest and size of the used file.
+//
+// This function can be used instead of dest.PutBlob() where the ImageDestination requires PutBlob() to be called.
+func PutBlobFromLocalFile(ctx context.Context, dest types.ImageDestination, file string) (digest.Digest, int64, error) {
```

Comment:
Let's make this more general by allowing the caller to pass a file descriptor instead, with a sugar wrapper method that opens for them if need be?

Or is this function *already* sugar basically and the functionality would already be exposed by `PutBlobWithOptions` as is?

Reply:
No, a file descriptor is a good idea — assuming Go has a good enough platform abstraction for it.

Reply:
On second thought, accepting a file descriptor brings up various questions, like “what does it mean if the file descriptor is positioned 1 GB into the file?”.

The use case we have now is to accept a path on the CLI, so a path-based interface is exactly what we need and avoids having to think about these things.

We can always add another function later.